src/pipx/venv.py

```diff
@@ -184,19 +263,130 @@ def get_python_version(self) -> str:
             .strip()
         )
 
+    def pip_search(self, search_term, pip_search_args):
+        cmd = [str(self.python_path), "-m", "pip", "search"]
+        cmd += pip_search_args + [search_term]
+        cmd_run = subprocess.run(cmd, stdout=subprocess.PIPE)
+        return cmd_run.stdout.decode().strip()
+
+    def list_installed_packages(self) -> Set[str]:
+        cmd = [str(self.python_path), "-m", "pip", "list", "--format=json"]
+        cmd_run = subprocess.run(cmd, stdout=subprocess.PIPE)
+        pip_list = json.loads(cmd_run.stdout.decode().strip())
+        return set([x["name"] for x in pip_list])
+
+    def top_of_deptree(self, packages):
+        top_package_name = ""
+
+        cmd = [str(self.python_path), "-m", "pip", "show"] + list(packages)
+        cmd_run = subprocess.run(cmd, stdout=subprocess.PIPE)
+        pip_show_stdout = cmd_run.stdout.decode().strip()
+
+        for line in pip_show_stdout.split("\n"):
+            key_value_re = re.search(r"^([^:]+):\s(.*)", line)
+            if key_value_re:
+                key = key_value_re.group(1)
+                value = key_value_re.group(2)
+                if key == "Name":
+                    package_name = value
+                if key == "Required-by" and value == "":
+                    top_package_name = package_name
```

Comment:
Why not just `return top_package_name` here?

Reply:
Sure.
Done.